geonode/people/utils.py

```diff
@@ -103,3 +106,19 @@ def format_address(street=None, zipcode=None, city=None, area=None, country=None
         if country:
             address.append(country)
         return " ".join(address)
+
+
+def get_available_users(user):
+    if user.is_superuser:
+        return get_user_model().objects.exclude(Q(username='AnonymousUser') | Q(is_active=False))
+    # Only return user that are members of any group profile the current user is member of
+    member_ids = list(GroupMember.objects.filter(
+        group__in=GroupProfile.objects.filter(
+            Q(access='public') | Q(group__in=user.groups.all()))
+        ).select_related('user').values_list('user__id', flat=True))
+    if Group.objects.filter(name=groups_settings.REGISTERED_MEMBERS_GROUP_NAME).count():
```

Comment:
Can we add a comment on why we always add the users that belong to the Registered member's groups? Just as an additional context

Reply:
updated

Reply:
updated